src/main/java/com/github/soerxpso/xpspawners/manager/SpawnerManager.java

```diff
@@ -10,57 +9,41 @@
 
 import com.github.soerxpso.xpspawners.Spawner;
 
+import vg.civcraft.mc.civmodcore.locations.QTBox;
+import vg.civcraft.mc.civmodcore.locations.SparseQuadTree;
+
 public class SpawnerManager {
-	private HashMap<Long, ArrayList<Spawner>> spawners = 
-			new HashMap<Long, ArrayList<Spawner>>();
+	
+	private SparseQuadTree spawners = new SparseQuadTree();
 	
 	public ItemStack convertSpawnerToItem(Block b) {
 		return new ItemStack(Material.MOB_SPAWNER);
 	}
 	
 	public Spawner nearestSpawner(Location loc) {
 		Spawner closest = null;
-		for(int xShift = -1; xShift <= 1; xShift++) {
-			for(int zShift = -1; zShift <= 1; zShift++) {
-				long chunkID = chunkID(loc.add(xShift, 0, zShift));
-				ArrayList<Spawner> chunk = spawners.get(chunkID);
-				if(chunk == null) continue;
-				for(Spawner s : chunk) {
-					if(closest == null) {
-						closest = s;
-						continue;
-					}
-					if(loc.distance(s.getLocation()) < loc.distance(closest.getLocation())) {
-						closest = s;
-					}
-				}
+		Set<QTBox> nearby = spawners.find(loc.getBlockX(), loc.getBlockZ());
```

Comment:
Find returns a set of all QTBoxes within range (a 33x33x33 cube). I'm looping through the set afterward to check which is nearest.